docs/subscriptions/index.md

```diff
@@ -0,0 +1,273 @@
+    );
+
+    /// <summary>
+    /// Register a callback to run when the subscription fails.
+    /// </summary>
+    public SubscriptionBuilder<SubscriptionEventContext, ErrorContext> OnError(
+        Action<ErrorContext, Exception> callback
+    );
+
+    /// <summary>
+    /// Subscribe to a subset of database via a single SQL query.
+    /// Returns a handle which you can use to monitor or cancel the subscription later.
+    /// </summary>
+    public SubscriptionHandle<SubscriptionEventContext, ErrorContext> Subscribe(
+        string querySql
+    );
+
+    /// <summary>
+    /// Subscribe to all rows from all tables.
+    ///
+    /// This method is intended as a convenience
+    /// for applications where client-side memory use and network bandwidth are not concerns.
+    /// Applications where these resources are a constraint
+    /// should register more precise queries via [`Subscribe`]
+    /// in order to replicate only the subset of data which the client needs to function.
+    ///
+    /// This method should not be combined with `Subscribe` on the same `DbConnection`.
+    /// A connection may either `Subscribe` to particular queries,
+    /// or `SubscribeToAllTables`, but not both.
+    /// Attempting to call `Subscribe`
+    /// on a `DbConnection` that has previously used `SubscribeToAllTables`,
+    /// or vice versa, may misbehave in any number of ways,
+    /// including dropping subscriptions or corrupting the client cache.
+    /// </summary>
+    public void SubscribeToAllTables();
+}
+```
+:::
+
+A `SubscriptionBuilder` provides an interface for registering subscription queries with a database.
+It allows you to register callbacks that run when the subscription is successfully applied or when an error occurs.
+Once applied, a client will start receiving row updates to its client cache.
+A client can react to these updates by registering row callbacks for the appropriate table.
+
+It is important to note that subscriptions must be disjoint from one another.
+Two subscriptions that return the same row may result in a corrupted client cache.
+That is, one that does not accurately and consistently reflect the state of the database.
+
+### Example Usage
+
+:::server-rust
+```rust
+// Establish a database connection
+let conn: DbConnection = connect_to_db();
+
+// Register a subscription with the database
+let subscription_handle = conn
+    .subscription_builder()
+    .on_applied(|ctx| { /* handle applied state */ })
+    .on_error(|error_ctx, error| { /* handle error */ })
+    .subscribe("SELECT * FROM my_table WHERE active = 1");
+```
+:::
+:::server-csharp
+```cs
+// Establish a database connection
+var conn = ConnectToDB();
+
+// Register a subscription with the database
+var userSubscription = conn
+    .SubscriptionBuilder()
+    .OnApplied((ctx) => { /* handle applied state */ })
+    .OnError((errorCtx, error) => { /* handle error */ })
+    .Subscribe("SELECT * FROM user");
+```
+:::
```

Comment:
I feel like the two langs should use the same query. The code should be as equivalent as possible.